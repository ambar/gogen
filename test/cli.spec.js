const {mock} = require('../lib')
const shell = require('../lib/utils/shell')

describe('gogen', () => {
  beforeAll(async () => {
    // install workspace deps
    await shell('yarn --silent', {cwd: 'examples'})
  })

  it('should throw errors', async () => {
    await expect(mock('', '')).rejects.toThrow()
    await expect(mock('foo', '')).rejects.toThrow()
    await expect(mock('', 'foo')).rejects.toThrow()
  })

  it('should generate `basic`', async () => {
    const {files, readFile} = await mock('./examples/basic', 'mylib')
    expect(files).toMatchSnapshot()

    // patches package.json
    expect(JSON.parse(readFile('package.json'))).toMatchObject({
      name: 'mylib',
      description: 'superb',
    })
    expect(JSON.parse(readFile('subpackage/package.json'))).toMatchObject({
      name: 'subpackage',
      license: 'MIT',
    })

    // re-run
    await expect(mock('./examples/basic', 'mylib')).resolves.not.toThrow()
  })

  it('should generate `with-ejs`', async () => {
    const {files, readFile} = await mock('./examples/with-ejs', 'mylib')
    expect(files).toMatchSnapshot()
    expect(JSON.parse(readFile('package.json'))).toMatchObject({
      description: 'superb',
    })
  })

  it('should generate `without-gogenrc`', async () => {
    const {files} = await mock('./examples/without-gogenrc', 'mylib')
    expect(files).toMatchSnapshot()
  })

  it('should generate `create-gogen`', async () => {
    const {files} = await mock('./examples/create-gogen', 'mylib')
    expect(files).toMatchSnapshot()
  })
})
