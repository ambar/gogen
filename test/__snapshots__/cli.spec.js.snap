// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gogen should generate \`basic\` 1`] = `
Array [
  ".eslintrc",
  ".gitignore",
  "README.md",
  "index.js",
  "package.json",
  "subpackage/.gitignore",
  "subpackage/package.json",
]
`;

exports[`gogen should generate \`create-gogen\` 1`] = `
Array [
  ".gitignore",
  ".gogenrc.js",
  "package.json",
  "template/gitignore",
  "template/index.js",
  "template/package.json",
]
`;

exports[`gogen should generate \`with-ejs\` 1`] = `
Array [
  ".gitignore",
  "README.md",
  "index.js",
  "package.json",
]
`;

exports[`gogen should generate \`with-prompts\` 1`] = `
Array [
  ".eslintrc",
  ".gitignore",
  "index.js",
  "package.json",
]
`;

exports[`gogen should generate \`without-gogenrc\` 1`] = `
Array [
  ".eslintrc",
  ".gitignore",
  "index.js",
  "package.json",
]
`;
